api/api.go

```diff
@@ -35,120 +30,115 @@ func HttpPOST(url string, data string) (resp *http.Response, err error) {
 			return
 		}
-		handler(w, req)
+		h.ServeHTTP(w, req)
 	})
 }
 
 // initAPI determines which functions handle each API call.
 func (srv *Server) initAPI() {
-	mux := http.NewServeMux()
-
-	// 404 Calls
-	srv.handleHTTPRequest(mux, "/", srv.unrecognizedCallHandler)
+	mux := httprouter.New()
+	mux.NotFound = http.HandlerFunc(srv.unrecognizedCallHandler) // custom 404
 
 	// Daemon API Calls - Unfinished
-	srv.handleHTTPRequest(mux, "/daemon/constants", srv.daemonConstantsHandler)
-	srv.handleHTTPRequest(mux, "/daemon/version", srv.daemonVersionHandler)
-	srv.handleHTTPRequest(mux, "/daemon/stop", srv.daemonStopHandler)
-	srv.handleHTTPRequest(mux, "/daemon/updates/apply", srv.daemonUpdatesApplyHandler)
-	srv.handleHTTPRequest(mux, "/daemon/updates/check", srv.daemonUpdatesCheckHandler)
+	mux.GET("/daemon/constants", srv.daemonConstantsHandler)
+	mux.GET("/daemon/version", srv.daemonVersionHandler)
+	mux.POST("/daemon/stop", srv.daemonStopHandler)
 
 	// Consensus API Calls
 	if srv.cs != nil {
-		srv.handleHTTPRequest(mux, "/consensus", srv.consensusHandler)            // GET
-		srv.handleHTTPRequest(mux, "/consensus/block", srv.consensusBlockHandler) // GET
+		mux.GET("/consensus", srv.consensusHandler)
+		mux.GET("/consensus/blocks/:height", srv.consensusBlocksHandler)
 	}
 
 	// Explorer API Calls
 	if srv.explorer != nil {
-		srv.handleHTTPRequest(mux, "/explorer", srv.explorerHandler)            // GET
-		srv.handleHTTPRequest(mux, "/explorer/", srv.explorerHandler)           // $(hash) GET
-		srv.handleHTTPRequest(mux, "/explorer/block", srv.explorerBlockHandler) // GET
+		mux.GET("/explorer", srv.explorerHandler)
+		mux.GET("/explorer/blocks/:height", srv.explorerBlocksHandler)
+		mux.GET("/explorer/hash/:hash", srv.explorerHashHandler)
 	}
 
 	// Gateway API Calls - Unfinished
 	if srv.gateway != nil {
-		srv.handleHTTPRequest(mux, "/gateway/status", srv.gatewayStatusHandler)
-		srv.handleHTTPRequest(mux, "/gateway/peers/add", srv.gatewayPeersAddHandler)
-		srv.handleHTTPRequest(mux, "/gateway/peers/remove", srv.gatewayPeersRemoveHandler)
+		mux.GET("/gateway", srv.gatewayHandler)
+		mux.POST("/gateway/add/:addr", srv.gatewayAddHandler)
+		mux.POST("/gateway/remove/:addr", srv.gatewayRemoveHandler)
 	}
 
 	// Host API Calls
 	if srv.host != nil {
-		srv.handleHTTPRequest(mux, "/host", srv.hostHandler)                  // GET, POST
-		srv.handleHTTPRequest(mux, "/host/announce", srv.hostAnnounceHandler) // POST
+		mux.GET("/host", srv.hostHandlerGET)
+		mux.POST("/host", srv.hostHandlerPOST)
+		mux.POST("/host/announce", srv.hostAnnounceHandler)
 	}
 
 	// HostDB API Calls - DEPRECATED
 	if srv.renter != nil {
-		srv.handleHTTPRequest(mux, "/hostdb/hosts/active", srv.renterHostsActiveHandler)
-		srv.handleHTTPRequest(mux, "/hostdb/hosts/all", srv.renterHostsAllHandler)
+		mux.HandlerFunc("GET", "/hostdb/hosts/active", srv.renterHostsActiveHandler)
+		mux.HandlerFunc("GET", "/hostdb/hosts/all", srv.renterHostsAllHandler)
 	}
```

Comment:
We do still need to be able to see a list of hosts, but I thought we were getting that from the renter now.